common/workunit/workunit.cpp

```diff
@@ -2988,10 +2971,20 @@ class CDaliWorkUnitFactory : public CWorkUnitFactory, implements IDaliClientShut
                 case WUStateDebugRunning:
                 case WUStateBlocked:
                 case WUStateAborting:
-                    SessionId agent = conn->queryRoot()->getPropInt64("@agentSession", -1);
-                    if (checkAbnormalTermination(wuid, ret, agent))
+                    if (queryDaliServerVersion().compare("2.1")>=0)
                     {
-                        return ret;
+                        SessionId agent = conn->queryRoot()->getPropInt64("@agentSession", -1);
+                        if((agent>0) && querySessionManager().sessionStopped(agent, 0))
+                        {
+                            conn->reload();
+                            // MCK - does checking state after sessionStopped reveal meaningful info ?
+                            WUState finalState = (WUState) getEnum(conn->queryRoot(), "@state", states);
+                            if ( (WUStateCompleted != finalState) &&
+                                 (WUStateFailed    != finalState) &&
+                                 (WUStateAborted   != finalState) )
```

Comment:
We would always break out of the loop for cases of WUStateCompiling, WUStateRunning, WUStateDebugPaused, WUStateDebugRunning, WUStateBlocked, WUStateAborting.
If we want to loop around again we would have to change the loop.  The 5.6.4 code seems right.